File: model/graduacao.js
import GraduacaoDAO from "../repository/graduacaoDAO.js";

export default class Graduacao {

    #id;
    #descricao;

    constructor(id = 0, descricao = 0) {
        this.#id = id;
        this.#descricao = descricao;
    }

    getId() {
        return this.#id;
    }

    setId(novoid) {
        this.#id = novoid;
    }

    getDescricao() {
        return this.#descricao;
    }

    setDescricao(novadescricao) {
        this.#descricao = novadescricao;
    }

    toJSON() {
        return {
            "id": this.#id,
            "descricao": this.#descricao,
        };
    }

    async gravar() {
        const graduacaoDAO = new GraduacaoDAO();
        return await graduacaoDAO.gravar(this);
    }

    async alterar() {
        const graduacaoDAO = new GraduacaoDAO();
        return await graduacaoDAO.alterar(this);
    }

    async apagar() {
        const graduacaoDAO = new GraduacaoDAO();
        return await graduacaoDAO.apagar(this.#id);
    }

    async consultar(filtro) {
        const graduacaoDAO = new GraduacaoDAO();
        return await graduacaoDAO.get(filtro);
    }
}

File: src/main/java/sistema/saga/sagabackend/model/Graduacao.java
package sistema.saga.sagabackend.model;

import sistema.saga.sagabackend.repository.Conexao;
import sistema.saga.sagabackend.repository.GraduacaoDAO;

import java.time.LocalDate;
import java.util.List;

public class Graduacao {
    private int id;
    private String descricao;

    public Graduacao() {
        this(0,"");
    }

    public Graduacao(String descricao) {
        this.descricao = descricao;
    }

    public Graduacao(int id, String descricao) {
        this.id = id;
        this.descricao = descricao;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public boolean apagar(Conexao conexao) {
        GraduacaoDAO graduacaoDAO = new GraduacaoDAO();
        return graduacaoDAO.apagar(this,conexao);
    }

    public int buscaGraduacao(Conexao conexao){
        GraduacaoDAO graduacaoDAO = new GraduacaoDAO();
        return graduacaoDAO.getGraduacao(this,conexao);
    }

    public List<Graduacao> buscarTodos(Conexao conexao) {
        GraduacaoDAO graduacaoDAO = new GraduacaoDAO();
        return graduacaoDAO.get("",conexao);
    }

    public boolean alterar(Conexao conexao) {
        GraduacaoDAO graduacaoDAO = new GraduacaoDAO();
        return graduacaoDAO.alterar(this,conexao);
    }

    public boolean gravar(Conexao conexao) {
        GraduacaoDAO graduacaoDAO = new GraduacaoDAO();
        return graduacaoDAO.gravar(this,conexao);
    }
}
